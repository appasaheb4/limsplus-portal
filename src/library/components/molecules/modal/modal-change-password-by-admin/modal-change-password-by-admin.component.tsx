import React, { useEffect } from 'react';
import { observer } from 'mobx-react';
import { Toast, List, Form } from '../../..';
import { FormHelper } from '@/helper';
import { Container } from 'reactstrap';
import { useForm, Controller } from 'react-hook-form';
import { useStores } from '@/stores';

interface ModalProps {
  show: boolean;
  title?: string;
  onClick: () => void;
  onClose: () => void;
}

export const ModalChangePasswordByAdmin = observer((props: ModalProps) => {
  const { userStore } = useStores();
  const {
    control,
    handleSubmit,
    formState: { errors },
    // setValue,
  } = useForm();
  const [showModal, setShowModal] = React.useState(props.show);

  const onSubmitModalChangePasswordByAdmin = () => {
    if (userStore.changePassword) {
      props.onClick();
    } else {
      Toast.error({
        message: 'ðŸ˜” Please enter all information!',
      });
    }
  };
  useEffect(() => {
    setShowModal(props.show);
  }, [props]);

  return (
    <Container>
      {showModal && (
        <>
          <div className='justify-center items-center flex overflow-x-hidden overflow-y-auto fixed inset-0 z-50 outline-none focus:outline-none'>
            <div className='relative w-full my-6 mx-auto max-w-3xl'>
              {/*content*/}
              <div className='border-0 rounded-lg shadow-lg relative flex flex-col w-full bg-white outline-none focus:outline-none p-3'>
                {/*header*/}
                <div className='flex items-start justify-between border-b border-solid border-gray-300 rounded-t '>
                  <div className='flex-col'>
                    <h3 className='text-3xl font-semibold'>
                      Change Password By Adminstrator
                    </h3>
                  </div>

                  <button
                    className='p-1  border-0 text-black opacity-1 ml-6 float-right text-3xl leading-none font-semibold outline-none focus:outline-none'
                    onClick={() => props.onClose && props.onClose()}
                  >
                    <span className=' text-black h-6 w-6 text-2xl block outline-none focus:outline-none'>
                      Ã—
                    </span>
                  </button>
                </div>

                {/*body*/}
                <div className='relative p-3 flex-auto'>
                  <List direction='col' space={4} justify='stretch' fill>
                    <Controller
                      control={control}
                      render={({ field: { onChange } }) => (
                        <Form.InputPassword
                          label='New Password'
                          name='newPassword'
                          hasError={!!errors.newPassword}
                          placeholder={
                            errors.newPassword
                              ? 'Please Enter New Password'
                              : 'New Password'
                          }
                          value={userStore.changePassword?.newPassword}
                          onChange={newPassword => {
                            onChange(newPassword);
                            userStore.updateChangePassword({
                              ...userStore.changePassword,
                              newPassword,
                            });
                          }}
                        />
                      )}
                      name='newPassword'
                      rules={{
                        required: true,
                        pattern: FormHelper.patterns.password,
                      }}
                      defaultValue=''
                    />
                    <Controller
                      control={control}
                      render={({ field: { onChange } }) => (
                        <Form.InputPassword
                          label='Confirm Password'
                          name='confirmPassword'
                          hasError={!!errors.confirmPassword}
                          placeholder={
                            errors.confirmPassword
                              ? 'Please Enter Confirm Password'
                              : 'Confirm Password'
                          }
                          value={userStore.changePassword?.confirmPassword}
                          onChange={confirmPassword => {
                            onChange(confirmPassword);
                            userStore.updateChangePassword({
                              ...userStore.changePassword,
                              confirmPassword,
                            });
                          }}
                        />
                      )}
                      name='confirmPassword'
                      rules={{
                        required: true,
                        pattern: FormHelper.patterns.password,
                        validate: value =>
                          value === userStore.changePassword?.newPassword,
                      }}
                      defaultValue=''
                    />
                  </List>
                </div>
                {/*footer*/}
                <div className='flex items-center justify-end  border-t border-solid border-gray-300 rounded-b p-2'>
                  <button
                    className='text-red-500 background-transparent font-bold uppercase px-6 py-2 text-sm outline-none focus:outline-none mr-1 mb-1'
                    type='button'
                    style={{ transition: 'all .15s ease' }}
                    onClick={() => props.onClose && props.onClose()}
                  >
                    Later
                  </button>
                  <button
                    className='bg-green-500 text-white active:bg-green-600 font-bold uppercase text-sm px-6 py-3 rounded shadow hover:shadow-lg outline-none focus:outline-none mr-1 mb-1'
                    type='button'
                    style={{ transition: 'all .15s ease' }}
                    onClick={handleSubmit(onSubmitModalChangePasswordByAdmin)}
                  >
                    Change
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div className='opacity-25 fixed inset-0 z-40 bg-black'></div>
        </>
      )}
    </Container>
  );
});
