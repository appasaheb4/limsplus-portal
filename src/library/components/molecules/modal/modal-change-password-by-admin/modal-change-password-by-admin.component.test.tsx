import React from 'react';
import {render} from '@testing-library/react';
import {ModalChangePasswordByAdmin} from './modal-change-password-by-admin.component';

it('render modalChangePasswordByAdmin correctly', () => {
  const modalChangePasswordByAdmin = render(
    <ModalChangePasswordByAdmin
      show={true}
      onClick={() => jest.fn()}
      onClose={() => jest.fn()}
    />,
  );
  expect(modalChangePasswordByAdmin).toMatchSnapshot();
});
