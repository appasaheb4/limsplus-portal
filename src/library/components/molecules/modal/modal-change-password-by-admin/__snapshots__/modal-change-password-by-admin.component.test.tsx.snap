// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalChangePassword component render modalChangePassword correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="justify-center items-center flex overflow-x-hidden overflow-y-auto fixed inset-0 z-50 outline-none focus:outline-none"
        >
          <div
            class="relative w-full my-6 mx-auto max-w-3xl"
          >
            <div
              class="border-0 rounded-lg shadow-lg relative flex flex-col w-full bg-white outline-none focus:outline-none p-3"
            >
              <div
                class="flex items-start justify-between border-b border-solid border-gray-300 rounded-t "
              >
                <div
                  class="flex-col"
                >
                  <h3
                    class="text-3xl font-semibold"
                  >
                    Change Password By Adminstrator
                  </h3>
                </div>
                <button
                  class="p-1  border-0 text-black opacity-1 ml-6 float-right text-3xl leading-none font-semibold outline-none focus:outline-none"
                >
                  <span
                    class=" text-black h-6 w-6 text-2xl block outline-none focus:outline-none"
                  >
                    ×
                  </span>
                </button>
              </div>
              <div
                class="relative p-3 flex-auto"
              >
                <span
                  class="flex flex-col space-y-4    flex-1 items-stretch"
                >
                  <div>
                    <label
                      class="text-gray-700 block text-3xs font-medium  mb-1"
                      for=""
                    >
                      New Password
                    </label>
                    <input
                      autocomplete="given-name"
                      class="undefined leading-4 p-2  focus:outline-none focus:ring  block w-full shadow-sm sm:text-base  border-2  border-gray-300 rounded-md"
                      data-testid="INPT"
                      name="newPassword"
                      placeholder="New Password"
                      type="password"
                      value=""
                    />
                  </div>
                  <div>
                    <label
                      class="text-gray-700 block text-3xs font-medium  mb-1"
                      for=""
                    >
                      Confirm Password
                    </label>
                    <input
                      autocomplete="given-name"
                      class="undefined leading-4 p-2  focus:outline-none focus:ring  block w-full shadow-sm sm:text-base  border-2  border-gray-300 rounded-md"
                      data-testid="INPT"
                      name="confirmPassword"
                      placeholder="Confirm Password"
                      type="password"
                      value=""
                    />
                  </div>
                </span>
              </div>
              <div
                class="flex items-center justify-end  border-t border-solid border-gray-300 rounded-b p-2"
              >
                <button
                  class="text-red-500 background-transparent font-bold uppercase px-6 py-2 text-sm outline-none focus:outline-none mr-1 mb-1"
                  style="transition: all .15s ease;"
                  type="button"
                >
                  Later
                </button>
                <button
                  class="bg-green-500 text-white active:bg-green-600 font-bold uppercase text-sm px-6 py-3 rounded shadow hover:shadow-lg outline-none focus:outline-none mr-1 mb-1"
                  style="transition: all .15s ease;"
                  type="button"
                >
                  Change
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="opacity-25 fixed inset-0 z-40 bg-black"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="justify-center items-center flex overflow-x-hidden overflow-y-auto fixed inset-0 z-50 outline-none focus:outline-none"
      >
        <div
          class="relative w-full my-6 mx-auto max-w-3xl"
        >
          <div
            class="border-0 rounded-lg shadow-lg relative flex flex-col w-full bg-white outline-none focus:outline-none p-3"
          >
            <div
              class="flex items-start justify-between border-b border-solid border-gray-300 rounded-t "
            >
              <div
                class="flex-col"
              >
                <h3
                  class="text-3xl font-semibold"
                >
                  Change Password By Adminstrator
                </h3>
              </div>
              <button
                class="p-1  border-0 text-black opacity-1 ml-6 float-right text-3xl leading-none font-semibold outline-none focus:outline-none"
              >
                <span
                  class=" text-black h-6 w-6 text-2xl block outline-none focus:outline-none"
                >
                  ×
                </span>
              </button>
            </div>
            <div
              class="relative p-3 flex-auto"
            >
              <span
                class="flex flex-col space-y-4    flex-1 items-stretch"
              >
                <div>
                  <label
                    class="text-gray-700 block text-3xs font-medium  mb-1"
                    for=""
                  >
                    New Password
                  </label>
                  <input
                    autocomplete="given-name"
                    class="undefined leading-4 p-2  focus:outline-none focus:ring  block w-full shadow-sm sm:text-base  border-2  border-gray-300 rounded-md"
                    data-testid="INPT"
                    name="newPassword"
                    placeholder="New Password"
                    type="password"
                    value=""
                  />
                </div>
                <div>
                  <label
                    class="text-gray-700 block text-3xs font-medium  mb-1"
                    for=""
                  >
                    Confirm Password
                  </label>
                  <input
                    autocomplete="given-name"
                    class="undefined leading-4 p-2  focus:outline-none focus:ring  block w-full shadow-sm sm:text-base  border-2  border-gray-300 rounded-md"
                    data-testid="INPT"
                    name="confirmPassword"
                    placeholder="Confirm Password"
                    type="password"
                    value=""
                  />
                </div>
              </span>
            </div>
            <div
              class="flex items-center justify-end  border-t border-solid border-gray-300 rounded-b p-2"
            >
              <button
                class="text-red-500 background-transparent font-bold uppercase px-6 py-2 text-sm outline-none focus:outline-none mr-1 mb-1"
                style="transition: all .15s ease;"
                type="button"
              >
                Later
              </button>
              <button
                class="bg-green-500 text-white active:bg-green-600 font-bold uppercase text-sm px-6 py-3 rounded shadow hover:shadow-lg outline-none focus:outline-none mr-1 mb-1"
                style="transition: all .15s ease;"
                type="button"
              >
                Change
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="opacity-25 fixed inset-0 z-40 bg-black"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
